src/test/java/mate/academy/bookstore/service/impl/BookServiceImplTest.java

```diff
@@ -0,0 +1,286 @@
+
+        List<BookDto> books = assertDoesNotThrow(() -> bookService.search(params, pageable));
+        assertEquals(EXPECTED_LIST_SIZE, books.size());
+    }
+
+    @Test
+    @DisplayName("Search books with negative price (ignore smaller max price)")
+    void search_ZeroMaxPrice_IgnoreSmallerMaxPrice() {
+        BookSearchParametersDto params = getSearchParams(NEGATIVE_PRICE, ZERO_MAX_PRICE);
+
+        Pageable pageable = PageRequest.of(PAGE_NUMBER, PAGE_SIZE);
+        BookDto bookDto = getBookDto(getCreateBookRequestDto());
+        Specification<Book> spec = (root, query, criteriaBuilder) -> criteriaBuilder.conjunction();
+
+        when(specificationBuilder.build(params)).thenReturn(spec);
+        when(bookRepository.findAll(spec, pageable)).thenReturn(new PageImpl<>(List.of(validBook)));
+        when(bookMapper.toDto(validBook)).thenReturn(bookDto);
+
+        List<BookDto> books = assertDoesNotThrow(() -> bookService.search(params, pageable));
+        assertEquals(EXPECTED_LIST_SIZE, books.size());
+    }
+
+    @Test
+    @DisplayName("Get all books by category id (return list of BookDtoWithoutCategoryIds)")
+    void getAllBookByCategoryId() {
+        Pageable pageable = PageRequest.of(PAGE_NUMBER, PAGE_SIZE);
+        BookDtoWithoutCategoryIds bookDtoWithoutCategoryIds = getBookDtoWithoutCategoryIds();
+
+        when(bookRepository.findAllBooksByCategoryId(VALID_CATEGORY_ID, pageable))
+                .thenReturn(List.of(validBook));
+        when(bookMapper.toDtoWithoutCategories(validBook)).thenReturn(bookDtoWithoutCategoryIds);
+
+        List<BookDtoWithoutCategoryIds> books =
+                bookService.getAllBookByCategoryId(VALID_CATEGORY_ID, pageable);
+
+        assertEquals(EXPECTED_LIST_SIZE, books.size());
+        assertEquals(bookDtoWithoutCategoryIds, books.getFirst());
+    }
+
+    private CreateBookRequestDto getCreateBookRequestDto() {
+        CreateBookRequestDto createBookDto = new CreateBookRequestDto();
+        createBookDto.setTitle(VALID_BOOK_TITLE);
+        createBookDto.setAuthor(VALID_BOOK_AUTHOR);
+        createBookDto.setIsbn(VALID_BOOK_ISBN);
+        createBookDto.setPrice(VALID_BOOK_PRICE);
+        createBookDto.setCategoryIds(SET_OF_VALID_CATEGORY_ID);
+        createBookDto.setDescription(VALID_BOOK_DESCRIPTION);
+        createBookDto.setCoverImage(VALID_BOOK_COVER_IMAGE);
+        return createBookDto;
+    }
+
+    private BookDtoWithoutCategoryIds getBookDtoWithoutCategoryIds() {
+        BookDtoWithoutCategoryIds bookDto = new BookDtoWithoutCategoryIds();
+        bookDto.setTitle(VALID_BOOK_TITLE);
+        bookDto.setAuthor(VALID_BOOK_AUTHOR);
+        bookDto.setIsbn(VALID_BOOK_ISBN);
+        bookDto.setPrice(VALID_BOOK_PRICE);
+        bookDto.setDescription(VALID_BOOK_DESCRIPTION);
+        bookDto.setCoverImage(VALID_BOOK_COVER_IMAGE);
+        return bookDto;
+    }
+
+    private BookDto getBookDto(CreateBookRequestDto dto) {
```

Comment:
it is not just the getter, it also has some parameter, and it essentially maps something to something